cmd/snap-confine/snap-confine-invocation.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SC_INVOCATION_H
+#define SC_INVOCATION_H
+
+#include <stdbool.h>
+
+/**
+ * sc_invocation contains information about how snap-confine was invoked.
+ *
+ * All of the pointer fields have the life-cycle bound to the main process.
+ **/
+typedef struct sc_invocation {
+	/* Things declared by the system. */
```

Comment:
are classic_confinement and executable going to be used soon?

Reply:
I would like to use them instead of the various locals in main(). The reason they are added here is that we can define sc_invocation as the external data about invoking snap-confine to run some program. It's the complete set.

Reply:
I have prepared tweak/refactor-sc-main-5 for that.